src/Common/AbstractApiRequest.php

```diff
@@ -31,6 +34,46 @@ public function getQueryParams(): string
         return $this->queryParams;
     }
 
+    public function getBasicAuthHeaderIfSet(): ?string
+    {
+        if (!$this->isBasicAuth()) {
+            return null;
+        }
+        return 'Basic ' . base64_encode(
+            "{$this->getBasicAuthUser()}:{$this->getBasicAuthPassword()}"
+        );
+    }
+
+    public function isBasicAuth(): bool
+    {
+        return $this->hasBasicAuth;
+    }
+
+    public function setHasBasicAuth($hasBasicAuth)
```

Comment:
Maybe throw some :void return types on these setters, assuming this won't be digested by any service lower than PHP 7.1